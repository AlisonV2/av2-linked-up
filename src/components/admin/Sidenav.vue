<template>
  <div class="col-auto col-md-3 col-xl-2 px-sm-2 px-0 bg-dark">
    <div
      class="
        d-flex
        flex-column
        align-items-center align-items-sm-start
        px-3
        pt-2
        text-white
        min-vh-100
      "
    >
      <router-link :to="{ name: 'Home' }" class="navbar-logo">
        <img class="admin-brand" src="/img/logo-light.png" alt="Brand logo" />
      </router-link>
      <ul
        class="
          nav nav-pills
          flex-column
          mb-sm-auto mb-0
          align-items-center align-items-sm-start
        "
        id="menu"
      >
        <li>
          <router-link
            :to="{ name: 'AdminProfile' }"
            class="nav-link align-middle px-0"
          >
            <i class="fs-4 bi-gear"></i>
            <span class="ms-3 d-none d-sm-inline">Profile</span>
          </router-link>
        </li>
        <hr class="dropdown-divider" />
        <li class="nav-item" v-if="role === 'artist'">
          <router-link
            :to="{ name: 'AdminGallery' }"
            class="nav-link align-middle px-0"
          >
            <i class="fs-4 bi-images"></i>
            <span class="ms-3 d-none d-sm-inline">Gallery</span>
          </router-link>
        </li>
        <hr class="dropdown-divider" v-if="role === 'artist'" />
        <li v-if="role === 'artist' || role === 'client'">
          <router-link
            :to="{ name: 'AdminProjects' }"
            class="nav-link align-middle px-0"
          >
            <i class="fs-4 bi-card-checklist"></i>
            <span class="ms-3 d-none d-sm-inline">Projects</span></router-link
          >
        </li>
        <hr
          class="dropdown-divider"
          v-if="role === 'artist' || role === 'client'"
        />
        <li>
          <router-link
            :to="{ name: 'AdminInbox' }"
            class="nav-link align-middle px-0"
          >
            <i class="fs-4 bi-envelope"></i>
            <span class="ms-3 d-none d-sm-inline">Messages</span></router-link
          >
        </li>
        <hr class="dropdown-divider" />
        <li>
          <router-link
            :to="{ name: 'AdminEvents' }"
            class="nav-link align-middle px-0"
          >
            <i class="fs-4 bi-calendar-date"></i>
            <span class="ms-3 d-none d-sm-inline">Events</span></router-link
          >
        </li>
        <template v-if="role === 'organizer'">
          <hr class="dropdown-divider" />
          <li v-if="role === 'organizer'">
            <router-link
              :to="{ name: 'ManageArtists' }"
              class="nav-link align-middle px-0"
            >
              <i class="fs-4 bi bi-card-checklist"></i>
              <span class="ms-3 d-none d-sm-inline">Artists</span></router-link
            >
          </li>
          <hr class="dropdown-divider" />
          <li>
            <router-link
              :to="{ name: 'ManageAttendees' }"
              class="nav-link align-middle px-0"
            >
              <i class="fs-4 bi bi-list-ol"></i>
              <span class="ms-3 d-none d-sm-inline"
                >Attendees</span
              ></router-link
            >
          </li>
        </template>
      </ul>
    </div>
  </div>
</template>

<script>
/**
 * @exports Sidenav
 * @type {Component}
 */
export default {
  name: 'Sidenav',
  data() {
    return {
      role: '',
    };
  },
  created() {
    this.$store.dispatch('getUserRole').then(() => {
      this.role = this.$store.getters.getUserRole;
    });
  },
};
</script>

<style lang="scss" scoped>
a,
.bi {
  color: $light;
}

a:hover,
.bi:hover,
a:active,
.bi:active {
  color: $primary;
  cursor: pointer;
}

.admin-brand {
  display: none;
}

@include bp-up(sm) {
  .dropdown-divider {
    border: 1px solid $light !important;
    width: 8rem;
  }
  .admin-brand {
    width: 8rem;
    transform: translate(1rem, -3rem);
    display: block;
  }
}
</style>
