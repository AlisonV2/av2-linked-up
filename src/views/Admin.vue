<template>
<div class="container-fluid">
    <div class="row flex-nowrap">
        <Sidenav />
        <div class="col py-3">
          <div class="container">
            <router-view></router-view>
          </div>
        </div>
    </div>
</div>
</template>

<script>
import Sidenav from '@/components/admin/Sidenav';
/**
 * @type Page
 * @name Admin
 */
export default {
  name: 'Admin',
  components: {
    Sidenav
  }
}
</script>