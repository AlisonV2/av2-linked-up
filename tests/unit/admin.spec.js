import { shallowMount } from '@vue/test-utils';
import Admin from '@/views/Admin.vue';
import Sidenav from '@/components/admin/Sidenav';

/**
 * @module AdminPageTest
 */
describe('Admin.vue', () => {
  /**
   * Check if Sidenav is rendered
   */
  it('Check if Sidenav is rendered', () => {
    const component = shallowMount(Admin, {
      global: {
        components: {
          Sidenav,
        },
        stubs: ['router-view'],
      },
    });
    const admin = component.findComponent(Sidenav);
    expect(admin.exists()).toBe(true);
  });
});
