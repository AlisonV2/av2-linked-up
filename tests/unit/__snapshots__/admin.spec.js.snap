// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin.vue AdminPage Snapshot 1`] = `
<div
  class="container-fluid"
>
  <div
    class="row flex-nowrap"
  >
    <sidenav-stub />
    <div
      class="col py-3"
    >
      <div
        class="container"
      >
        <router-view-stub />
      </div>
    </div>
  </div>
</div>
`;
